myftp.py

```diff
@@ -115,5 +115,61 @@ def extractPort(data):
             response = control_socket.recv(1024).decode('utf-8').strip()
             print(response)
 
+            data_socket = socket(AF_INET, SOCK_STREAM)
+            dataPASV = extractPASVData(response)
+            ip = extractIP(dataPASV)
+            port = extractPort(dataPASV)
+        elif command == 'ls':
+            if len(listOfArgs) != 1:
+                print('Put command must be: ls')
+            else:
+                filename = listOfArgs
+                control_socket.send(bytes('PASV\r\n', 'utf-8'))
+                response = control_socket.recv(1024).decode('utf-8').strip()
+                print(response)
+
+                data_socket = socket(AF_INET, SOCK_STREAM)
+                dataPASV = extractPASVData(response)
+                ip = extractIP(dataPASV)
+                port = extractPort(dataPASV)
+
+                try:
+                    data_socket.connect((ip, port))
+                except Exception:
+                    print(f'Error: server {ip} cannot be found.')
+                else:
+                    print(f'Connected to data socket: {ip}')
+                    control_socket.send(bytes(f'LIST\r\n', 'utf-8'))
+                    response = control_socket.recv(1024).decode('utf-8').strip()
+                    print(response)
+                    response = data_socket.recv(1024).decode('utf-8').strip()
+                    print(response)
+                    data_socket.close()
+        elif command == 'cd':
+            if len(listOfArgs) != 2:
+                print('Put command must be: cd <name of file>')
+            else:
+                nameFile = listOfArgs[1]

```

Comment:
```suggestion
                directoryName = listOfArgs[1]

```
Should change variable name to reflect better what it's storing.